# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 30d2bf04719b
# Full Hash: 30d2bf04719bfb60095aaa7d679363e57d25ad92
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-06-13 03:21:52
# Regressor Bug: 1640019
# File Overlap Count: 1
# Description:
#   Bug 1640019 - Part 1: Support toplevel process switches outside of tabbrowser, r=mattwoodrow
#   
#   This new process switching behavior is only enabled for some browser elements,
#   which have specified a specific attribute. Turning this on for all browsers with
#   a `remote` attribute causes breakage in reftests.
# ==============================================================================

diff -r 3fefa1d023a6 -r 30d2bf04719b dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Fri Jun 12 17:02:45 2020 +0000
+++ b/dom/ipc/ContentParent.cpp	Fri Jun 12 16:52:00 2020 +0000
@@ -1316,7 +1316,7 @@
   RefPtr<ContentParent> constructorSender;
   MOZ_RELEASE_ASSERT(XRE_IsParentProcess(),
                      "Cannot allocate BrowserParent in content process");
-  if (aOpenerContentParent) {
+  if (aOpenerContentParent && aOpenerContentParent->IsAlive()) {
     constructorSender = aOpenerContentParent;
   } else {
     if (aContext.IsJSPlugin()) {